### DISCUSSION TITLE
Re-train or not ?

### DISCUSSION THREAD

**Discussion by Ze-213-33** [kyleskom/NBA-Machine-Learning-Sports-Betting#133] [Q&A]
Hi 
Do I need to re-train the models with new data or just clone the directory and do the inferences. 
Here is the version of the results I got today, are they coherent according to you please ?
Also is there any tutorial to understand the code ?

---------------XGBoost Model Predictions---------------
Washington Wizards (56.5%) vs Atlanta Hawks: OVER 236 (49.9%)
Miami Heat vs Cleveland Cavaliers (54.1%): OVER 215 (51.0%)
Boston Celtics (71.7%) vs Portland Trail Blazers: UNDER 230.5 (61.0%)
New Orleans Pelicans (56.7%) vs Dallas Mavericks: UNDER 234.5 (54.1%)
Denver Nuggets (79.2%) vs Chicago Bulls: OVER 225.5 (61.2%)
Phoenix Suns (63.9%) vs Oklahoma City Thunder: UNDER 233.5 (50.3%)
LA Clippers (57.8%) vs Toronto Raptors: UNDER 230 (56.2%)

**Comment by kyleskom** [comment]
You don't need to retrain the model.

**Reply by inar-vision** [reply]
Hi! Just found this repo and I'm trying it out. Why are the "train the model" instructions made available, if I don't need to train it? Not trying to be rude, just trying to understand. :)

**Reply by kyleskom** [reply]
1.) I have the train the model at some point so the code lives with the repo
2.) You / others can still train models if you want to play around with it but isn't needed for daily predictions.

**Reply by inar-vision** [reply]
Check!! Thanks!

**Reply by inar-vision** [reply]
Just to make things clear for myself, it is required that I run the Get_Data & Get_Odds_Data before I run main.py?

**Reply by kyleskom** [reply]
No, I'll train the models and have the best models always being used.

**Reply by inar-vision** [reply]
Well that's just sweet. Thank you my man!

**Comment by Skysafe64** [comment]
Interesting, I ran the script at around the same time and didn't get the same results, I thought the predictions didn't move based on the odds? 
![image](https://user-images.githubusercontent.com/123671403/223899740-ce194d2b-0b6c-455c-bb84-180d59090f25.png)

**Reply by kyleskom** [reply]
Expected value will change based on changes in odds.

**Reply by Skysafe64** [reply]
> Expected value will change based on changes in odds.

however it also changes the prediction percentage of over/under and ML.
